src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddress.cs

```diff
@@ -121,11 +121,19 @@ private static bool TryParse(string address, string? displayName, Encoding? disp
         {
             if (address == null)
             {
-                throw new ArgumentNullException(nameof(address));
+                if (throwExceptionIfFail)
+                    throw new ArgumentNullException(nameof(address));
+
+                parsedData = default;
+                return false;
             }
             if (address.Length == 0)
             {
-                throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
+                if (throwExceptionIfFail)
+                    throw new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
+
+                parsedData = default;
+                return false;
             }
```

Comment:
This could be simplified to avoid the redundant code, e.g.
```C#
if (string.IsNullOrEmpty(address))
{
    if (throwExceptionIfFail)
    {
        throw address is null ?
            new ArgumentNullException(nameof(address)) :
            new ArgumentException(SR.Format(SR.net_emptystringcall, nameof(address)), nameof(address));
    }

    parsedData = default;
    return false;
}
```